pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -568,11 +569,22 @@ pub mod pallet {
 		///   pool.
 		/// - `Error::<T>::PoolNotLive`: If the pool is not in a live (locked or
 		///   active) state.
+		/// - `Error::<T>::InvalidInput`: If all lock flags are `true` (all
+		///   operations enabled), which would be equivalent to an unlocked
+		///   (active) pool state.
 		#[pallet::call_index(3)]
 		#[pallet::weight(T::WeightInfo::set_lock())]
 		pub fn set_lock(origin: OriginFor<T>, pool_id: T::PoolId, lock: Locks) -> DispatchResult {
 			let who = T::DefaultOrigin::ensure_origin(origin)?;
 
+			ensure!(
+				lock != (Locks {
```

Comment:
Can we do the same without allocating a new `Locks` object?

Reply:
Good point, that's bad. I wanted to avoid the scenario where we add a field to the locks and then forget adding a check for this new field here, but I think fa4bde344f88c9f4274eab98c8aa743c03cbd0c2 does avoid that pretty well too